### PR TITLE
Schedule jobs for creating release PR from mgmt. SDK release request

### PR DESCRIPTION
**Pull request by Alancere**

### CODE REVIEW

**Comment by weshaggard** [comment]
@Alancere @tadelesh 
Can you give more details on this pipeline? I want to make sure I understand the goals of the auto-release and how it is expected to be triggered?

**Comment by tadelesh** [comment]
@weshaggard This pipeline aims to create release [PR](https://github.com/Azure/azure-sdk-for-go/pull/18525) from the release request [issue](https://github.com/Azure/sdk-release-request/issues?q=is%3Aopen+is%3Aissue+label%3AGo). It is a daily manual work for now and we want to use pipeline to schedule do it automatically. This approach is similar with Python's [automation](https://github.com/Azure/azure-sdk-for-python/blob/main/scripts/release_helper/release_helper.yml) in use.

**Comment by weshaggard** [comment]
Thanks for the context. @lirenhe how do you see this fitting in with our new sdk release powerapps flow? Do you see this still be a necessary step or would we attempt another release process?

File: eng/pipelines/templates/jobs/auto-release.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+trigger:
+  branches:
+    exclude:
+      - '*'
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/auto-release.yml:4]:
```suggestion
trigger: none
```

File: eng/pipelines/templates/jobs/auto-release.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+# avoid being triggered as part of CI check
+pr:
+  branches:
+    exclude:
+      - '*'
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/auto-release.yml:10]:
```suggestion
pr: none
```

File: eng/pipelines/templates/jobs/auto-release.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+schedules:
+  - cron: "0,30 1-9 * * *"
+    displayName: Daily release
+    branches:
+      include:
+        - main
+    always: true
+
+variables:
+  - group: Release Secrets for GitHub
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/auto-release.yml:21]: We generally try to avoid the variable group name in the yml file but instead just set it up in the DevOps UI.

**Comment by weshaggard** [comment]
@dw511214992 should this process be handled by the SDK Automation instead of independent pipeline? It feels to me that we should be creating these release PR's automatically has spec changes get merged.

**Comment by tadelesh** [comment]
> @dw511214992 should this process be handled by the SDK Automation instead of independent pipeline? It feels to me that we should be creating these release PR's automatically has spec changes get merged.

Currently, we release our SDK only when there is release request. Spec change will be too frequent for SDK release. Is there any manually release trigger for the new powerapps workflow?

**Comment by dw511214992** [comment]
I agree with @tadelesh . SDK automation pipeline cannot be used for release now because it's triggered by swagger PR. This PR is target at releasing go sdk by querying release request automatically.

When the powerapp is GA, I don't we need this pipeline. But for now, it's worth to create the pipeline as powerapp is not ready.

**Comment by weshaggard** [comment]
Thanks for the extra context. I'm fine with using this as a release helper until we get a better E2E solution for releasing.

File: eng/pipelines/templates/jobs/auto-release.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            
+              
+              file_size=`du -b ./sdk-release.json |awk '{print $1}'`
+              echo "sdk-release.json file size:" ${file_size}
+              if [ ${file_size} -le 70 ]; then
+                echo "There are no services that need to be released"
+              else
+                echo "run generator release-v2..."
+                generator release-v2 ./azure-sdk-for-go ./azure-rest-api-specs ./sdk-release.json -t $(azuresdk-github-pat)
+              fi
```
Comment (tadelesh) [inline eng/pipelines/templates/jobs/auto-release.yml:63]:
```suggestion
          - script: |
              export PATH=$PATH:$HOME/go/bin
              sudo ln -s /usr/bin/pwsh /usr/bin/pwsh.exe
              git config --global user.email "ReleaseHelper"
              git config --global user.name "ReleaseHelper"
              
              git clone https://github.com/Azure/azure-sdk-for-go.git
              git clone https://github.com/Azure/azure-rest-api-specs.git
              
              cd azure-sdk-for-go
              git remote add fork https://Azure:$(azuresdk-github-pat)@github.com/Azure/azure-sdk-for-go.git
              cd ../
              
              go install github.com/Azure/azure-sdk-for-go/eng/tools/generator@latest
              
              generator issue -t $(azuresdk-github-pat) > sdk-release.json
              cat sdk-release.json
            
              
              file_size=`du -b ./sdk-release.json |awk '{print $1}'`
              echo "sdk-release.json file size:" ${file_size}
              if [ ${file_size} -le 70 ]; then
                echo "There are no services that need to be released"
              else
                echo "run generator release-v2..."
                generator release-v2 ./azure-sdk-for-go ./azure-rest-api-specs ./sdk-release.json -t $(azuresdk-github-pat)
              fi
```

File: eng/pipelines/mgmt-auto-release.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+              command: 'custom'
+              customCommand: 'install -g autorest@latest'
+
+          - task: GoTool@0
+            inputs:
+              version: '1.18.2'
+
+          - script: |
+              export PATH=$PATH:$HOME/go/bin
+              sudo ln -s /usr/bin/pwsh /usr/bin/pwsh.exe
```
Comment (benbp) [inline eng/pipelines/mgmt-auto-release.yml:30]: Why is this necessary? Are there tools in this pipeline that only look for `pwsh.exe`? Can we improve them so they are more cross platform compatible?

File: eng/pipelines/mgmt-auto-release.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+              git config --global user.email "ReleaseHelper"
+              git config --global user.name "ReleaseHelper"              
+              
+              cd ../
+              git clone https://github.com/Azure/azure-sdk-for-go.git
+              git clone https://github.com/Azure/azure-rest-api-specs.git
+              
+              cd azure-sdk-for-go
+              git remote add fork https://Azure:$(azuresdk-github-pat)@github.com/Azure/azure-sdk-for-go.git
```
Comment (benbp) [inline eng/pipelines/mgmt-auto-release.yml:39]: Why `fork` out of curiosity?

**Review by benbp — APPROVED** [review]

File: eng/pipelines/mgmt-auto-release.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          - task: Npm@1
+            inputs:
+              command: 'custom'
+              customCommand: 'install -g autorest@latest'
+
+          - task: GoTool@0
+            inputs:
+              version: '1.18.2'
+
+          - script: |
```
Comment (benbp) [inline eng/pipelines/mgmt-auto-release.yml:28]: Ideally this would get broken out into its own script so changes in functionality can be more easily tested.

File: eng/pipelines/mgmt-auto-release.yml (hunk 6 above)
Comment (Alancere) [reply eng/pipelines/mgmt-auto-release.yml:39]: In this [PR](https://github.com/Azure/azure-sdk-for-go/pull/18455) we changed the generator tools and added 'git remote fork' to facilitate submitting PRs from the personal fork repository to the origin repository

File: eng/pipelines/mgmt-auto-release.yml (hunk 7 above)
Comment (Alancere) [reply eng/pipelines/mgmt-auto-release.yml]: Moved scripts section to /eng/scripts/mgmt-auto-release.sh

**Comment by Alancere** [comment]
Results of the: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1748064&view=logs&j=0ab14b9f-e499-56d5-97b1-fd98b70ea339&t=2e9e19b8-1db7-53d9-fe91-457c791ee9df
